test/foundry/deploy/15_holograph_registry_tests.t.sol

```diff
@@ -0,0 +1,802 @@
+    //         }
+
+/**
+ * @notice Test the return of valid holographable contracts from the registry.
+ * @dev This test verifies that the registry returns a list of valid holographable contracts.
+ * It checks that the length of the returned list matches the expected count and that the list includes a specific contract address.
+ * Refers to the hardhat test with the description 'Should return valid contracts'
+ */
+    function testReturnValidHolographableContract() public {
+        uint16 expectedHolographableContractsCount = 5;
+        address[] memory contracts = registry.getHolographableContracts(0, expectedHolographableContractsCount);
+        assertEq(contracts.length, expectedHolographableContractsCount);
+
+    //     ERC721ConfigParams memory params = ERC721ConfigParams({
+    //         network: "Localhost",
+    //         deployer: deployer,
+    //         contractName: "SampleERC721",
+    //         collectionName: "Sample ERC721 Contract (Localhost)",
+    //         collectionSymbol: "SMPLR",
+    //         royaltyBps: 1000,
+    //         eventConfig: '0x0000000000000000000000000000000000000000000000000000000000000087',
+    //         initCode: abi.encode(deployer),
+    //         salt: "0x0000000000000000000000000000000000000000000000000000000000001000"
+    //         });
+
+    // (DeploymentConfig memory erc721Config, bytes32 erc721ConfigHash, bytes32 erc721ConfigHashBytes) = generateErc721Config(params);
+
+        bool found = false;
+        for (uint i = 0; i < contracts.length; i++) {
+            if (contracts[i] == address(sampleErc721Holographer)) {
+                found = true;
+                break;
+            }
+        }
+        assertTrue(found, "Expected contract address not found");
+    }
+
+/**
+ * @notice Test allowing an external contract to call the getHolographableContracts function.
+ * @dev This test verifies that an external contract can successfully call the getHolographableContracts function
+ * of the registry contract with the parameters 0 and 1.
+ * Refers to the hardhat test with the description 'Should allow external contract to call fn'
+ */
+    function testAllowExternalContractToCallGetHolographableContracts() public {
+        bytes memory encodeSignature = abi.encodeWithSignature('getHolographableContracts(uint256,uint256)', 0 , 1);
+        mockExternalCall.callExternalFn(address(registry), encodeSignature);
+        
+    }
+
+/*
+    getHolographableContractsLength
+*/
+
+/**
+ * @notice Test the validity of the length of holographable contracts returned by the registry.
+ * @dev This test verifies that the length of the list of holographable contracts returned by the registry
+ * matches the expected count of 5.
+ * Refers to the hardhat test with the description 'Should return valid _holographableContracts length'
+ */    
+    function testReturnValid_holographableContractsLength() public {
+        uint16 expectedHolographableContractsCount = 5;
```

Comment:
Where does this number come from, does it come from the deployment script?
I see it in the test above, could you move it to the top?